src/commands/buffer.rs

```diff
@@ -126,7 +126,11 @@ impl Buffer {
             bail!("Invalid size for buffer: {}", size);
         }
 
+        let len = self.data_buffer.len();
```

Comment:
Shouldn't we check the capacity of the `Vec` rather than its current length?

Reply:
Yeah, that would be better. I'll change it.